lxd/device/disk.go

```diff
@@ -1035,3 +1047,63 @@ func (d *disk) getParentBlocks(path string) ([]string, error) {
 
 	return devices, nil
 }
+
+// generateVMConfigDrive generates an ISO containing the cloud init config for a VM.
+// Returns the path to the ISO.
+func (d *disk) generateVMConfigDrive() (string, error) {
+	scratchDir := filepath.Join(d.instance.DevicesPath(), strings.Replace(d.name, "/", "-", -1))
+
+	// Create config drive dir.
+	err := os.MkdirAll(scratchDir, 0100)
+	if err != nil {
+		return "", err
+	}
+
+	instanceConfig := d.instance.ExpandedConfig()
+
+	// Use an empty user-data file if no custom vendor-data supplied.
+	vendorData := instanceConfig["user.vendor-data"]
+	if vendorData == "" {
+		vendorData = "#cloud-config"
+	}
+
+	err = ioutil.WriteFile(filepath.Join(scratchDir, "vendor-data"), []byte(vendorData), 0400)
+	if err != nil {
+		return "", err
+	}
+
+	// Use an empty user-data file if no custom user-data supplied.
+	userData := instanceConfig["user.user-data"]
+	if userData == "" {
+		userData = "#cloud-config"
+	}
+
+	err = ioutil.WriteFile(filepath.Join(scratchDir, "user-data"), []byte(userData), 0400)
+	if err != nil {
+		return "", err
+	}
+
+	metaData := fmt.Sprintf(`instance-id: %s
+local-hostname: %s
```

Comment:
This doesn't match what we do in 9p, the user can set `user.meta-data` to have content appended to those two lines.